docs/source/api/apollo-gateway.mdx

```diff
@@ -67,35 +80,48 @@ The core of the federated gateway implementation. For an example, see the ["impl
 
   * `debug`: `Boolean`
 
-    With debug enabled, the server will log startup messages as well as query plans during incoming requests.
+    If `true`, the gateway logs startup messages, along with the query plan for
+    each incoming request. The default value is `false`.
 
 #### Returns
 
-An `ApolloGateway` instance, which is intended to be passed as the `gateway` configuration option of `ApolloServer` like so:
+An `ApolloGateway` instance, which you then pass as the `gateway` configuration option to the `ApolloServer` constructor, like so:
 
 ```javascript
 const server = new ApolloServer({
   gateway: new ApolloGateway({ serviceList: [...] }),
 });
 ```
 
-## RemoteGraphQLDataSource
+## `RemoteGraphQLDataSource`
 
-The `RemoteGraphQLDataSource` export is used to connect to underlying federated services from the gateway. It allows for customizations of its request using the `willSendRequest` function. By default, the `ApolloGateway` creates a new `RemoteGraphQLDataSource` for each service using the `url` from the service definition as its target location.
+A `RemoteGraphQLDataSource` object represents a connection between your
+federated gateway and a single implementing service. You can customize this
+communication by defining a `willSendRequest` function that lets you modify each
+request's structure before it's sent.
 
 ### `constructor(options)`: `RemoteGraphQLDataSource`
 
 #### Parameters
 
 * `options`: `Object`
 
+  An object that contains configuration options for the data source.
+
+  Supported options include:
+
   * `url`: `string` _(required)_
 
-    The location of the service to fetch using an HTTP request.
+    The implementing service's URL for sending fetch requests via HTTP.
 
   * `willSendRequest`: `(request: GraphQLRequest) => Promise<void>`
```

Comment:
The type of `request` is incorrect here - from the code:

```ts
  public willSendRequest?<TContext>(
    requestContext: Pick<
      GraphQLRequestContext<TContext>,
      'request' | 'context'
    >,
  ): ValueOrPromise<void>;
```

...which translates to an object containing a `request` and `context` key.

Reply:
I don't know enough about how we're defining these signatures in the docs to get this change right; I'll catch up with you about it on Monday :D

Reply:
We might consider implying more directly that these URLs are, by default, communicated with over HTTP, but the connection layer can be overridden to support any protocol :)